### PR TITLE
Add flags to enable strict stable API

### PR DESCRIPTION
**Pull request by b1ron**
Closes FerretDB/dance/issues/791.

### CODE REVIEW

**Review by AlekSi — CHANGES_REQUESTED** [review]
> We should use appropriate _standard_ libraries

Not third-party dependencies

File: dotnet-example.csproj [hunk 1]
```diff
@@ -9,6 +9,7 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.Extensions.CommandLineUtils" Version="1.1.1" />
```
Comment (AlekSi) [inline dotnet-example.csproj:12]: Please use https://learn.microsoft.com/en-us/dotnet/standard/commandline/

File: Program.cs [hunk 2]
```diff
@@ -1,39 +1,66 @@
...
-
-        var filter = Builders<BsonDocument>.Filter.Eq("a", 4);
-        BsonDocument actual = collection.Find(filter).FirstOrDefault();
-        Debug.Assert(actual == new BsonDocument { { "_id", 4 }, { "a", 4 } }, "Value should be 4");
-
-        // prevents https://jira.mongodb.org/browse/CSHARP-3429
-        client.Cluster.Dispose();
+        var connectionString = new Argument<string>(
+        name: "connectionString",
+        description: "MongoDB connection string");
```
Comment (AlekSi) [inline Program.cs:12]: I don't think we should introduce a third style of indentation/brackets in the same file

File: Program.cs [hunk 3]
```diff
@@ -1,39 +1,66 @@
...
+        name: "--strict",
+        description: "Use strict stable API mode.");
+        strict.AddAlias("-s");
+        strict.SetDefaultValue(false);
+
+        var rootCommand = new RootCommand("A simple example of using MongoDB with .NET Core");
+        rootCommand.AddArgument(connectionString);
+        rootCommand.AddOption(strict);
+
+        rootCommand.SetHandler((connectionString, strict) =>
```
Comment (AlekSi) [inline Program.cs:24]: We could extract that nested anonymous function into its own method

Comment (b1ron) [reply Program.cs:24]: Sure

**Review by AlekSi — APPROVED** [review]

**Pull request merged**